File: Java/PA3/DictionaryTest.java
/* Samer Baslan
 * sbaslan
 * CMPS 12B
 * 07/23/2017
 * DictionaryTest (Test client for Dictionary ADT)
 * PA3
 */


class DictionaryTest{
	public static void main(String[] args){
		Dictionary A = new Dictionary();
		//System.out.println(A.isEmpty());
		A.insert("2", "c");
		//System.out.println(A);
		A.insert("1", "a");
		A.insert("3", "b");
		A.insert("5", "d");
		//System.out.println(A);
		//A.delete("5", "d"); exception
		A.delete("5");
		//System.out.println(A);
		//System.out.println(A.size());
		//System.out.println(A.isEmpty());
		//System.out.println(A.size());
		//A.insert("6"); exception 
		//A.insert("2", "e"); duplicate key exception
		//System.out.println(A);
		A.makeEmpty();
		A.insert("1", "a");
		A.insert("2", "b");
		A.insert("3", "c");
		A.insert("4", "c");
		//System.out.println(A);
		//System.out.println(A.lookup("3"));
		//System.out.println(A.findKey("2"));
		System.out.println(A.size());
		System.out.println(A.toString());

	}

}

File: C/Lab5/DictionaryTest.c
//Samer Baslan
//sbaslan
//CMPS 12M
//08-06-2017
//DictionaryTest.c
//Test client for Dictionary ADT

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
	Dictionary A = newDictionary();
	char* k;
	char* v;
	char* word1[] = {"t", "e", "s", "t", "1"};
	char* word2[] = {"g", "o", "o", "d", "2"};
	int i;

	for( i = 0; i < 4; i++){
	insert(A, word1[i], word2[i]);
	}
	
//	printDictionary(stdout, A);

	for( i = 0; i < 5; i++){
	k = word1[i];
	v = lookup(A, k);
	printf("key=\"%s\"%s\"%s\"\n", k, (v==NULL?"not found":"value="), v);
	}

	//delete(A, "2");
	//delete(A, "4");

	//printDictionary(stdout, A);
	
	//insert(A, "key2", "val2");
	//printDictionary(stdout, A);

	printf("%s\n", (isEmpty(A)?"true":"false"));
	printf("%d\n", size(A));
	makeEmpty(A);
	printf("%s\n", (isEmpty(A)?"true":"false"));

	freeDictionary(&A);
	return(EXIT_SUCCESS);
}
